napari/utils/colormaps/colormap_utils.py

```diff
@@ -581,125 +583,126 @@ def ensure_colormap(colormap: ValidColormapArg) -> Colormap:
         If a dict is provided and any of the values are not Colormap instances
         or valid inputs to the Colormap constructor.
     """
-    with AVAILABLE_COLORMAPS_LOCK:
```

Comment:
If we want the same thread-safety guarantees as before, at a minimum I'd expect to see the lock held while making the copy. Unless `copy` is guaranteed to be an atomic operation?

Reply:
After a quick look, allowing the rest of the code to run concurrently could result in some changes in behavior. For example, [`_increment_unnamed_colormap`](https://github.com/napari/napari/pull/5662/files#diff-93b06ecc55258074223c21034a86e7fbaaffe5157565c04c336baec8260a7777R608) could be interleaved now so could result in a name collision.

Still trying to understand if that matters with your other changes though.

Reply:
Ah, now I get it. After this PR, `AVAILABLE_COLORMAPS` is never mutated in napari's code. So you don't need any lock to protect against interleaved read/writes of that dict (because there are no writes).

My only concern is that `AVAILABLE_COLORMAPS` is part of napari's public API, directly from this module and from `napari.utils.colormaps`. That is style as if it's a (deeply immutable) constant, even though it isn't, so maybe no-one is directly mutating it?